incuna-sass/mixins/_em-scale.sass

```diff
@@ -23,30 +26,14 @@
 
 // You can also use variables to reference media query variables in your
 // project:
-//     @include em-scale(14px, $px768, 16px, $px1025, 20px)
+//     @include em-scale(14px, (($px768, 16px), ($px1025, 20px)))
 // or
-//     @include em-scale(14px, $tablet, 16px, $desktop, 20px)
+//     @include em-scale(14px, (($tablet, 16px), ($desktop, 20px)))
 
-
-@mixin em-scale($font-size-1, $breakpoint-1, $font-size-2, $breakpoint-2: false, $font-size-3: false, $breakpoint-3: false, $font-size-4: false, $breakpoint-4: false, $font-size-5: false, $breakpoint-5: false, $font-size-6: false)
+@mixin em-scale($font-base, $breakpoints-sizes-list: ())
     font:
-        size: $font-size-1
-    @include media($breakpoint-1)
-        font:
-            size: $font-size-2
-    @if $breakpoint-2
-        @include media($breakpoint-2)
-            font:
-                size: $font-size-3
-    @if $breakpoint-3
-        @include media($breakpoint-3)
-            font:
-                size: $font-size-4
-    @if $breakpoint-4
-        @include media($breakpoint-2)
-            font:
-                size: $font-size-3
-    @if $breakpoint-5
-        @include media($breakpoint-3)
+        size: $font-base
+    @each $change in $breakpoints-sizes-list
+        @include media(#{nth($change, 1)})
             font:
-                size: $font-size-4
+                size: nth($change, 2)
```

Comment:
>`$list-item`

👍  perfect name

> also I didn't really think the variables added anything in terms of readability, just an extra level of abstraction which seemed unnecessary.

I was convinced to have meaningful variable names by @wytrych, and I'm a total convert now.

The advantage they give you is their intention is obvious from their name; you don't have to derive it from the context, or `git blame` to consult the dev and find out the intention for the code. It also helps with debugging.

Example: if the order of the items in a `$list-item` is reversed, it will break. However, `$breakpoint: nth($list-item, 1)` immediately tells us that the first item of each `$list-item` is _meant_ to be a breakpoint string, so we don't have to figure that out. Without it, we have to go through an extra level of debugging, which may involve the question
> Do Sass lists start at `0` instead? Because it looks like `1` is getting me the font-size when it should be the second item.

Since we're writing more and more functional Sass, I think we should agree on a common style that can be applied across languages. I highly recommend reading the Clean Code chapter on variables: http://www.itiseezee.com/?p=83 =)